### PR TITLE
#166790049 Implements password reset feature

### PR DESCRIPTION
**Pull request by mnzube**
#### What does this PR do?
Implements password resetting and updating to a new password for AH users
#### Description of Task to be completed?
-create modal for the reset password
-Introduce actions and reducers
-Write password reset logic in the component folder
-Write password update logic in the component folder
-Introduce routes for the above components
-Introduce page styling
#### How should this be manually tested?
-Clone the repository locally with git clone https://github.com/andela/ah-jawans-frontend.git command.
-Check out to this branch by running git checkout ft-passwordReset-166790049.
-Install dependencies by running `npm install`.
-Update the .env file according to the .env.sample file
-Start the server with `npm start` and put this URL http://localhost:8080/passwordReset and http://localhost:8080/updatePassword in the address bar of your browser then follow the instruction of resetting your password.
#### What are the relevant pivotal tracker stories?
[#166790049](https://www.pivotaltracker.com/story/show/166790049)
#### Screenshot?
password

### CODE REVIEW

File: src/__tests__/components/PasswordReset.test.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    const fakeEvent = { preventDefault: () => {} };
+    const instance = component.instance();
+    instance.handleSubmit(fakeEvent);
+    expect(instance).toBeDefined();
+  });
+
+  it('should change states using handleChange method', () => {
+    const fakeEvent = { target: { name: 'email', value: 'yes dady' } };
+    const instance = component.instance();
+    console.log('>>>>>>>>>>', props.requestToReset.reset);
```
Comment (Joe-Joseph) [inline src/__tests__/components/PasswordReset.test.js]: Please remove the console.log

File: src/__tests__/components/PasswordReset.test.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    instance.handleChange(fakeEvent);
+    expect(instance).toBeDefined();
+  });
+
+  it('should find the modal', () => {
+    const newProps = { ...props, resetPassword: { errors: '' } };
+    const components = shallow(<PasswordReset {...newProps} />);
+
+    components.setState({ modal: {} });
+    // console.log('>>>>>......', components.debug());
```
Comment (Joe-Joseph) [inline src/__tests__/components/PasswordReset.test.js]: Same here

File: src/__tests__/redux/actions/ActionPasswordReset.test.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { requestPasswordRequest, changePassword } from '../../../redux/actions/passwordReset';
+import * as types from '../../../redux/actions/types';
+
+describe('actions', () => {
+  it('should create an action to login a user', () => {
+    const payload = {
+      email: 'joe@gmail.com',
+      password: 'Joe@123',
+    };
+    // console.log('>>>><<<</////', requestPasswordRequest(payload));
```
Comment (Joe-Joseph) [inline src/__tests__/redux/actions/ActionPasswordReset.test.js]: Same here

File: src/__tests__/components/PasswordReset.test.js (hunk 1 above)
Comment (mnzube) [reply src/__tests__/components/PasswordReset.test.js]: removed it Joe

**Review by pngabo — APPROVED** [review]

**Review by Joe-Joseph — APPROVED** [review]
Good work

**Pull request merged**